lib/renderer/extensions/web-navigation.ts

```diff
@@ -1,18 +1,16 @@
-'use strict'
-
-const Event = require('@electron/internal/renderer/extensions/event')
+import Event from '@electron/internal/renderer/extensions/event'
 const { ipcRendererInternal } = require('@electron/internal/renderer/ipc-renderer-internal')
 
 class WebNavigation {
-  constructor () {
-    this.onBeforeNavigate = new Event()
-    this.onCompleted = new Event()
+  private onBeforeNavigate = new Event()
+  private onCompleted = new Event()
 
-    ipcRendererInternal.on('CHROME_WEBNAVIGATION_ONBEFORENAVIGATE', (event, details) => {
+  constructor () {
+    ipcRendererInternal.on('CHROME_WEBNAVIGATION_ONBEFORENAVIGATE', (event: Event, details: any) => {
```

Comment:
This is gonna use the wrong `Event` so I'm surprised TS allowed it.  You shouldn't have to define the types here, TS should infer them from `electron.d.ts`
```suggestion
    ipcRendererInternal.on('CHROME_WEBNAVIGATION_ONBEFORENAVIGATE', (event, details) => {
```

Reply:
Same below

Reply:
I assumed this too but it blows up with 

```
details has implicit type 'any'
```
etc